File: packages/cli/src/commands/versions.ts
import { NexumClient } from 'nexum-js';

const workflowId = process.argv[3];
if (!workflowId) {
  console.error('Usage: nexum versions <workflow-id>');
  process.exit(1);
}

const client = new NexumClient();
const versions = await client.listWorkflowVersions(workflowId);

if (versions.length === 0) {
  console.log(`No versions found for "${workflowId}".`);
} else {
  console.log(`\nVersions for "${workflowId}":\n`);
  for (const v of versions) {
    const marker = v.activeExecutions > 0 ? ' <- ACTIVE' : '';
    console.log(`  ${v.versionHash.slice(0, 16)}... [${v.compatibility}] ${v.registeredAt}${marker}`);
    if (v.activeExecutions > 0) {
      console.log(`    ${v.activeExecutions} execution(s) in progress`);
    }
  }
}

File: cmd/nexum/cmd/versions.go
package cmd

import (
	"context"
	"fmt"
	"time"

	ngrpc "github.com/kuro6061/nexum/cmd/nexum/grpc"
	pb "github.com/kuro6061/nexum/cmd/nexum/proto"
	"github.com/spf13/cobra"
)

var versionsCmd = &cobra.Command{
	Use:   "versions <workflow_id>",
	Short: "List workflow versions",
	Args:  cobra.ExactArgs(1),
	RunE:  runVersions,
}

func init() {
	rootCmd.AddCommand(versionsCmd)
}

func runVersions(cmd *cobra.Command, args []string) error {
	workflowID := args[0]

	client, conn, err := ngrpc.Connect(serverAddr)
	if err != nil {
		return err
	}
	defer conn.Close()

	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
	defer cancel()

	resp, err := client.ListWorkflowVersions(ctx, &pb.ListVersionsRequest{
		WorkflowId: workflowID,
	})
	if err != nil {
		return fmt.Errorf("ListWorkflowVersions failed: %w", err)
	}

	if len(resp.Versions) == 0 {
		fmt.Printf("No versions found for workflow %s.\n", workflowID)
		return nil
	}

	fmt.Printf("WORKFLOW: %s\n", workflowID)
	for _, v := range resp.Versions {
		status := "INACTIVE"
		if v.ActiveExecutions > 0 {
			status = "ACTIVE"
		}
		deployed := formatTimeAgo(v.RegisteredAt)
		fmt.Printf("  %-16s %-10s deployed %s\n", v.VersionHash, status, deployed)
	}

	return nil
}
